collated/main/quanle1994.md

```diff
@@ -0,0 +1,745 @@
+# quanle1994
+###### /java/seedu/address/logic/commands/LoginCommand.java
+``` java
+
+/**
+ * Log the user in.
+ */
+public class LoginCommand extends Command {
+
+    public static final String COMMAND_WORD = "login";
+    public static final String COMMAND_ALIAS = "li";
+    public static final String MESSAGE_SUCCESS = "Log In Successful";
+    private static final String MESSAGE_ERROR_NO_USER = "User does not exist";
+    private static final String MESSAGE_ERROR_WRONG_PASSWORD = "Wrong Password";
+    private byte[] password;
+    private String userId;
+    private String passwordText;
+
+    public LoginCommand(String userId, String passwordText) {
+        this.userId = userId;
+        this.passwordText = passwordText;
+    }
+
+    public static String getCommandWord() {
+        return COMMAND_WORD;
+    }
+
+    private boolean isSameDigest(byte[] digest1, byte[] digest2) {
+        return Arrays.equals(digest1, digest2);
+    }
+
+    @Override
+    public CommandResult execute() throws CommandException {
+        requireNonNull(model);
+        if (!checkExistingUserId()) {
+            return new CommandResult(MESSAGE_ERROR_NO_USER);
+        }
+        String pwSalt = null;
+        try {
+            pwSalt = getSalt();
+        } catch (UserNotFoundException e) {
+            e.printStackTrace();
+        }
+        String combinedPw = pwSalt + passwordText;
+        if (!matchedPassword(new HashDigest().getHashDigest(combinedPw))) {
+            return new CommandResult(MESSAGE_ERROR_WRONG_PASSWORD);
+        }
+        return new CommandResult(MESSAGE_SUCCESS);
+    }
+
+    private boolean matchedPassword(byte[] digest) {
+        return isSameDigest(password, digest);
+    }
+
+    /**
+     * This checks if the userId is existing
+     */
+    private boolean checkExistingUserId() {
+        byte[] uidDigest = new HashDigest().getHashDigest(userId);
+        //byte[] retrievedDigest = model.retrieveDigestFromStorage();
```

Comment:
Don't leave commented code if you have no use for it, as we are using revision control, so we can get the history of changes when we need it.